externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/ExecuteStatement.scala

```diff
@@ -67,9 +70,23 @@ class ExecuteStatement(
         iter =
           if (incrementalCollect) {
             info("Execute in incremental collect mode")
-            new IterableFetchIterator(resultSetWrapper.toIterable)
+            new IterableFetchIterator[Row](new Iterable[Row] {
+              private val fetched = new AtomicBoolean(false)
+              override def iterator: Iterator[Row] = {
+                if (fetched.getAndSet(true)) {
+                  jdbcStatement.synchronized {
```

Comment:
we need to re-execute the statement when resetPosition

Reply:
Would re-execution be a possible problem? For example, insert is executed twice.

Reply:
For update statement `hasResult` should be `false`.

https://github.com/apache/kyuubi/blob/74416e935e565a760b09ddc259365d73e05aeb85/externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/ExecuteStatement.scala#L66